specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/runCommands.json

```diff
@@ -446,6 +735,160 @@
+          "description": "Specifies the script content to be executed on the VM."
+        },
+        "scriptUri": {
+          "type": "string",
+          "description": "Specifies the scrit download location."
+        }
+      },
+      "description": "Describes the script sources for run command."
+    },
+    "VirtualMachineRunCommandProperties": {
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/VirtualMachineRunCommandScriptSource",
+          "description": "The source of the run command script."
+        },
+        "parameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RunCommandParameterDefinition"
+          },
+          "description": "The parameters used by the script."
+        },
+        "protectedParameters": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/RunCommandParameterDefinition"
+          },
+          "description": "The parameters used by the script."
+        },
+        "runAsUser": {
+          "type": "string",
+          "description": "Specifies the user account on the VM when executing the run command."
+        },
+        "runAsPassword": {
+          "type": "string",
+          "description": "Specifies the user account password on the VM when executing the run command."
+        },
+        "timeoutInSeconds": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The timeout in seconds to execute the run command."
+        },
+        "outputBlobUri": {
+          "type": "string",
+          "description": "Specifies the Azure storage blob where script output stream will be uploaded."
+        },
+        "errorBlobUri": {
+          "type": "string",
+          "description": "Specifies the Azure storage blob where script error stream will be uploaded."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The provisioning state, which only appears in the response."
+        },
+        "instanceView": {
+          "readOnly": true,
+          "$ref": "#/definitions/VirtualMachineRunCommandInstanceView",
+          "description": "The virtual machine run command instance view."
+        }
+      },
+      "description": "Describes the properties of a Virtual Machine run command."
+    },
+    "VirtualMachineRunCommand": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineRunCommandProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./compute.json#/definitions/Resource"
+        }
+      ],
+      "description": "Describes a Virtual Machine run command."
+    },
+    "VirtualMachineRunCommandUpdate": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineRunCommandProperties"
```

Comment:
You have the same VirtualMachineRunCommandProperties for both PUT and PATCH, which mean, if any mandatory property of PUT is also mandatory in PATCH.

Reply:
yes